packages/frontend/src/hooks/cartesianChartConfig/useCartesianChartConfig.ts

```diff
@@ -432,46 +458,50 @@ const useCartesianChartConfig = ({
         );
     }, [resultsData?.metricQuery.dimensions, itemsMap, columnOrder]);
 
-    const [
-        availableFields,
-        availableDimensions,
-        availableMetrics,
-        availableTableCalculations,
-        availableCustomDimensions,
-    ] = useMemo(() => {
-        const metrics = resultsData?.metricQuery.metrics || [];
-        const tableCalculations =
-            resultsData?.metricQuery.tableCalculations.map(
-                ({ name }) => name,
-            ) || [];
-        const customDimensions =
-            resultsData?.metricQuery.customDimensions?.map(
-                getCustomDimensionId,
-            ) || [];
-        return [
-            [
-                ...sortedDimensions,
-                ...metrics,
-                ...tableCalculations,
-                ...customDimensions,
-            ],
-            [...sortedDimensions, ...customDimensions],
-            metrics,
-            tableCalculations,
-            customDimensions,
-        ];
-    }, [resultsData, sortedDimensions]);
+    const [availableFields, availableDimensions, availableMetrics] =
+        useMemo(() => {
+            const metrics = resultsData?.metricQuery.metrics || [];
+            const tableCalculations =
+                resultsData?.metricQuery.tableCalculations.map(
+                    ({ name }) => name,
+                ) || [];
+            const customDimensions =
+                resultsData?.metricQuery.customDimensions?.map(
+                    getCustomDimensionId,
+                ) || [];
+            return [
+                [
+                    ...sortedDimensions,
+                    ...metrics,
+                    ...tableCalculations,
+                    ...customDimensions,
+                ],
+                [...sortedDimensions, ...customDimensions],
+                metrics,
+            ];
+        }, [
+            resultsData?.metricQuery.customDimensions,
+            resultsData?.metricQuery.metrics,
+            resultsData?.metricQuery.tableCalculations,
+            sortedDimensions,
+        ]);
 
     // Set fallout layout values
     // https://www.notion.so/lightdash/Default-chart-configurations-5d3001af990d4b6fa990dba4564540f6
     useEffect(() => {
-        if (availableFields.length > 0) {
+        if (availableFields.length > 0 || columnOrder.length > 0) {
             setDirtyLayout((prev) => {
                 const isCurrentXFieldValid: boolean =
                     prev?.xField === EMPTY_X_AXIS ||
-                    (!!prev?.xField && availableFields.includes(prev.xField));
+                    (!!prev?.xField &&
+                        (availableFields.includes(prev.xField) ||
+                            columnOrder.includes(prev.xField)));
```

Comment:
could we add a comment here explaining why it could be done on `availableFields` or in `columnOrder`?